tar.go

```diff
@@ -61,6 +61,17 @@ func (*Tar) CheckExt(filename string) error {
 	return nil
 }
 
+// CheckPath ensures that the filename has not been crafted to perform path traversal attacks
+func (*Tar) CheckPath(to, filename string) error {
+	to, _ = filepath.Abs(to) //explicit the destination folder to prevent that 'string.HasPrefix' check can be 'bypassed' when no destination folder is supplied in input
+	dest := filepath.Join(to, filename) 
+	//prevent path traversal attacks
+	if !strings.HasPrefix(dest, to) {
+		return fmt.Errorf("illegal file path: %s", filename)
+	}
+	return nil
+}
```

Comment:
It seems this functionality is duplicated, better to extract to a separate function and call it.

Might be better to define it on `struct{}` and inline into formats.